tests/app/dao/test_monthly_billing.py

```diff
@@ -0,0 +1,126 @@
+
+    create_notification(template=sample_template, created_at=datetime(2017, 1, 15), billable_units=2,
+                        status='delivered')
+    create_notification(template=sample_template, created_at=datetime(2017, 1, 17, 13, 30, 57), billable_units=4,
+                        status='delivered')
+
+    create_or_update_monthly_billing_sms(service_id=sample_template.service_id,
+                                         billing_month=rate_2)
+    monthly_billing = MonthlyBilling.query.all()
+    assert len(monthly_billing) == 1
+    assert monthly_billing[0].month == 'January'
+
+    january = get_monthly_billing_sms(service_id=sample_template.service_id, billing_month=rate_2)
+    first_row = {"billing_units": 2,
+                 "rate_multiplier": 1,
+                 "international": False,
+                 "rate": 0.0158,
+                 "total_cost": 3 * 0.0158}
+    assert_monthly_billing(january, 2017, "January", sample_template.service_id, 2, first_row)
+    second_row = {"billing_units": 6,
+                  "rate_multiplier": 1,
+                  "international": False,
+                  "rate": 0.0124,
+                  "total_cost": 1 * 0.0124}
+    assert sorted(january.monthly_totals[1]) == sorted(second_row)
+
+
+def test_update_monthly_billing_overwrites_old_totals(sample_template):
+    july = datetime(2017, 7, 1)
+    create_rate(july, 0.123, 'sms')
+    create_notification(template=sample_template, created_at=datetime(2017, 7, 2), billable_units=1, status='delivered')
+
+    create_or_update_monthly_billing_sms(sample_template.service_id, july)
+    first_update = get_monthly_billing_sms(sample_template.service_id, july)
+    expected = {"billing_units": 1,
+                "rate_multiplier": 1,
+                "international": False,
+                "rate": 0.123,
+                "total_cost": 1 * 0.123}
+    assert_monthly_billing(first_update, 2017, "July", sample_template.service_id, 1, expected)
+
+    create_notification(template=sample_template, created_at=datetime(2017, 7, 5), billable_units=2, status='delivered')
+    create_or_update_monthly_billing_sms(sample_template.service_id, july)
+    second_update = get_monthly_billing_sms(sample_template.service_id, july)
+    expected_update = {"billing_units": 3,
+                       "rate_multiplier": 1,
+                       "international": False,
+                       "rate": 0.123,
+                       "total_cost": 3 * 0.123}
+    assert_monthly_billing(second_update, 2017, "July", sample_template.service_id, 1, expected_update)
+
+
+def assert_monthly_billing(monthly_billing, year, month, service_id, expected_len, first_row):
+    assert monthly_billing.year == year
+    assert monthly_billing.month == month
+    assert monthly_billing.service_id == service_id
+    assert len(monthly_billing.monthly_totals) == expected_len
+    assert sorted(monthly_billing.monthly_totals[0]) == sorted(first_row)
+
+
+def test_get_service_id():
```

Comment:
when using the `create_*` functions, an unfortunate downside is that they don't automatically trigger the db cleanup. You'll need to add `notify_db_session` as a parameter to this function. 

The others are fine since they still have `sample_template`, which requires `notify_db_session`